flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -114,77 +128,100 @@ private ScanContext(boolean caseSensitive, Long snapshotId, Long startSnapshotId
     this.includeColumnStats = includeColumnStats;
     this.exposeLocality = exposeLocality;
     this.planParallelism = planParallelism;
+
+    validate();
+  }
+
+  private void validate() {
+    if (isStreaming) {
+      if (startingStrategy == StreamingStartingStrategy.SPECIFIC_START_SNAPSHOT_ID) {
+        Preconditions.checkArgument(startSnapshotId != null,
+            "startSnapshotId cannot be null for SPECIFIC_START_SNAPSHOT_ID starting strategy");
```

Comment:
We probably want to remove the specific variable names from the error messages. And we also phrase error messages more directly: `(Problem): (context)`

This should probably be: `Invalid starting snapshot for SPECIFIC_START_SNAPSHOT_ID strategy: null`

Reply:
will update